streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBPrefixIterator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.rocksdb.RocksIterator;
+
+import java.nio.ByteBuffer;
+import java.util.Set;
+
+class RocksDBPrefixIterator extends RocksDbIterator {
+    private final byte[] rawPrefix;
+
+    RocksDBPrefixIterator(final String name,
+                          final RocksIterator newIterator,
+                          final Set<KeyValueIterator<Bytes, byte[]>> openIterators,
+                          final Bytes prefix) {
+        super(name, newIterator, openIterators, true);
+        this.rawPrefix = prefix.get();
+        newIterator.seek(rawPrefix);
+    }
+
+    private boolean prefixEquals(final byte[] prefix1, final byte[] prefix2) {
+        final int min = Math.min(prefix1.length, prefix2.length);
+        final ByteBuffer prefix1Slice = ByteBuffer.wrap(prefix1, 0, min);
+        final ByteBuffer prefix2Slice = ByteBuffer.wrap(prefix2, 0, min);
+        return prefix1Slice.equals(prefix2Slice);
```

Comment:
With this code `abcd` would be a prefix of `abc`. Is this intended? Those cases should be tested in the missing unit tests for this class.

Reply:
Ticket for unit test cases for RocksDBRangeIterator: https://issues.apache.org/jira/browse/KAFKA-10766

Reply:
I see what you mean. But why do you bother then to find the minimum length of the two prefixes? You could just use the length of the first prefix.

Reply:
Yeah I have actually simplified the logic. Don't think I needed prefix1 and prefix2 as one of the prefixes is always fixed(i.e the prefix we are searching for).